tools/toolchains/__init__.py

```diff
@@ -702,6 +719,7 @@ def get_arch_file(self, objects):
 
     # THIS METHOD IS BEING CALLED BY THE MBED ONLINE BUILD SYSTEM
     # ANY CHANGE OF PARAMETERS OR RETURN VALUES WILL BREAK COMPATIBILITY
+    @check_toolchain_path
```

Comment:
perhaps this could be named more general way because it might be that you want to check inc_dirs and other thing here as well before start compiling in future,,,,